### ISSUE TITLE
Asynchronous method in a conditional attribute notation

### ISSUE THREAD

**Issue by vdegenne** [Polymer/lit-html#544]
I am trying to use the conditional attribute notation with an `async` function, something like that :
```html
<span ?special="${this.asyncMethodThatReturnsTrueOrFalse()}">...</span>
```
However It seems that *lit-html* will always evaluate this expression as `true` even if the method returns `false` ([minimal example](https://glitch.com/edit/#!/lit-html-async-conditional-attribute)).

**Comment by justinfagnani** [comment]
Thanks for the reproduction @vdegenne!

**Comment by bgotink** [comment]
At the moment only `NodePart`s handle promises and other types, `AttributePart`, `PropertyPart`, `EventPart` and `BooleanAttributePart` take the value as-is without special handling for promises, iterables, nodes etc.

In my opinion that shouldn't change. For boolean attributes, changing the behaviour would give unexpected consequences when trying to set an attribute depending on whether a value is truthy or not. That'll work for any value, except for promises (or rather: any value with a `then` property) because for those we'll look at the truthiness of the value they eventually get fulfilled with.  
It's of course true that one can always write

```html
<span ?special=${this.somePromise != null}>
```

which will always work with or without the change. When I see a template with 

```html
<span ?special=${this.something}>
```

I interpret that as 

```js
if (this.something) {
  spanEl.setAttribute('special', '');
} else {
  spanEl.removeAttribute('special');
}
```

---

We could easily update the `until` directive to support other parts and not just `NodePart`s. Right now this directive makes use of the fact that `NodePart` supports promises out of the box, but we could easily update `until` to subscribe to the promise itself.

This feels more correct to me for two reasons:

1. I explicitly declare that I want to use the async value of the promise, not the fact that the promise is present
1. This allows me to give a default value. If `BooleanAttributePart` automatically handles promises, I cannot declare whether I want the attribute to be present or not while the promise is unresolved.

**Comment by justinfagnani** [comment]
Closed by #555
